elastalert/create_index.py

```diff
@@ -5,25 +5,53 @@
 import json
 import os
 import time
+import yaml
+import requests
 
 import elasticsearch.helpers
-import yaml
+
 from elasticsearch import RequestsHttpConnection
 from elasticsearch.client import Elasticsearch
 from elasticsearch.client import IndicesClient
 from elasticsearch.exceptions import NotFoundError
 from envparse import Env
 
 from elastalert.auth import Auth
-from elastalert.util import get_version_from_cluster_info
+from elastalert.util import get_version_from_cluster_info, is_true, is_response_ok
 
 env = Env(ES_USE_SSL=bool)
 
+def create_quickwit_mappings(client_infos, recreate):
+    qw_index_mappings = read_qw_index_mappings()
+    for index_id in qw_index_mappings:
+        if not recreate:
+            r = requests.get("{}/api/v1/indexes/{}/describe".format(client_infos['url'], index_id), auth=client_infos['http_auth'], headers=client_infos['headers'])
+            if is_response_ok(r.status_code):
+                print('Index ' + index_id + ' already exists. Skipping index creation.')
+                continue
+
+        r = requests.post("{}/api/v1/indexes".format(client_infos['url']), json=qw_index_mappings[index_id], auth=client_infos['http_auth'], headers=client_infos['headers'])
+        if is_response_ok(r.status_code):
+            print('Index ' + index_id + ' successfully created.')
+        else:
+            print('Index ' + index_id + ' not created because of error. Try to delete it and retry again...')
```

Comment:
Should say: `... error. Attempting to recreate index.`.

Reply:
Done